apps/whois/mode/whois.pm

```diff
@@ -0,0 +1,308 @@
+#
+# Copyright 2021 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors : Pedro Manuel Santos Delgado - lunik
+#
+
+package apps::whois::mode::whois;
+
+use strict;
+use warnings;
+use base qw(centreon::plugins::mode);
+use DateTime;
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $self->{version} = '1.0';
+    $options{options}->add_options(arguments => {
+        "domain:s"            => { name => 'domain'},
+        "whois_binary:s"      => { name => 'command', default => '/usr/bin/whois'},
+        "timeout:s"           => { name => 'timeout', default => 3},
+        "warning-days:s"      => { name => 'warning', default => "30:"},
+        "critical-days:s"     => { name => 'critical', default => "10:"},
+        "custom-date-regex:s" => { name => 'custom_date_regex', default => 'Expiry Date:\s+(\d{2})-(\d{2})-(\d{4})'}, # Regex for whois outcom
+        "custom-date-map:s"   => { name => 'custom_date_map', default => "DMY"}, # day month year
+    });
+    # This method will execute the command, but will not let me parse the outcome
+    return $self;
+}
+
```

Comment:
Add your counters definition here. 

For example you'll have a global/days counter, type 0. 

See a full example of something similar here: https://github.com/centreon/centreon-plugins/blob/master/apps/protocols/x509/mode/certificate.pm